src/infrastructure/api/index.ts

```diff
@@ -0,0 +1,112 @@
+import testQueryFunctionFactory from '../../app/testQueryFunctionFactory';
+import { getResultsDetails, getVehicleDetails } from '../../domain/enquiryService';
+import ParametersError from '../../errors/ParametersError';
+import ResultsEvent from '../../interfaces/ResultsEvent';
+import VehicleEvent from '../../interfaces/VehicleEvent';
+import DatabaseService from '../databaseService';
+import SecretsManagerService from '../secretsManagerService';
+
+const app = express();
+const router = express.Router();
+
+const { API_VERSION } = process.env;
+
+// Declare middlewares
+/**
+ * bodyParser, error handling, logger, etc..
+ * http://expressjs.com/en/starter/basic-routing.html
+ * http://expressjs.com/en/guide/using-middleware.html
+ */
+
+/**
+ * app level middlewares
+ * app.use('/path', (req, res, next) => {
+ * chain middlewares
+ * next()
+ * })
+ */
+router.use((req, _response, next) => {
+  // TODO Add logger lib like Winston or Morgan
+  console.log('path');
+  console.log(req.path);
+  next();
+});
+
+// Debug router before we start proxying  requests from /v<x> psth
+router.get('/', (_request, res) => {
+  res.send({ ok: true });
+});
+
+router.get('/version', (_request, res) => {
+  res.send({ version: API_VERSION });
+});
+
+router.get(
+  '/vehicle',
+  (
+    request: Request<Record<string, unknown>, string | Record<string, unknown>, Record<string, unknown>, VehicleEvent>,
+    res,
+  ) => {
+    console.info('Handling vehicle request');
+    let secretsManager;
+    try {
+      secretsManager = new SecretsManagerService(new AWS.SecretsManager());
+    } catch (e) {
+      if (e instanceof Error) {
+        res.status(500).send(e.message);
+      }
+    }
+
+    DatabaseService.build(secretsManager, mysql)
+      .then((dbService) => getVehicleDetails(request.query, vehicleQueryFunctionFactory, dbService))
+      .then((result) => {
+        res.contentType('json').send(result);
+      })
+      .catch((e: Error) => {
+        if (e instanceof ParametersError) {
+          res.status(400);
+        } else {
+          res.status(500);
+        }
+
+        res.send(e.message);
+      });
+  },
+);
+
+router.get(
+  '/testResults',
+  (
+    request: Request<Record<string, unknown>, string | Record<string, unknown>, Record<string, unknown>, ResultsEvent>,
+    res,
+  ) => {
+    const secretsManager = new SecretsManagerService(new AWS.SecretsManager());
+
+    DatabaseService.build(secretsManager, mysql)
+      .then((dbService) => getResultsDetails(request.query, testQueryFunctionFactory, dbService))
+      .then((result) => {
+        res.contentType('json').send(JSON.stringify(result));
+      })
+      .catch((e: Error) => {
+        if (e instanceof ParametersError) {
+          res.status(400);
+        } else {
+          res.status(500);
+        }
+
+        res.send(e.message);
+      });
+  },
+);
+
+router.all('*', (_request, res) => {
+  res.status(405).send();
```

Comment:
I am thinking that don't we want to send 404 instead of 405 if no route is found so the user knows that he is using the api in a wrong way (no route available)? 405 is telling the user that a method is not allowed where for a same resource we could allow it or not? Will we have RAC user permissions later?

Reply:
I've changed this so that they get a 405 if they try to access one of the endpoints with the wrong type of request (POST, DELETE, etc.) and 404 if they access a completely incorrect route.

Reply:
Thanks for explaining. I am not sure what the initial requirement was for that and if we were going to test it as I couldn't find it defined in the ticket